### ISSUE TITLE
c++17 structured bindings

### ISSUE THREAD

**Issue by niekbouman** [root-project/cling#193]
Hello,

It seems that c++17 structured bindings do not yet work in cling.
Is this indeed the case, or did I setup cling incorrectly ? 

```cpp
auto t = make_tuple("hi",3);
auto [a,b] = t;
```

best regards,
niek

**Comment by Axel-Naumann** [comment]
I never tried. Before I do :-) did you build cling with C++17 enabled?

**Comment by niekbouman** [comment]
Yup, I am using cling-cpp17 kernel. For example, C++17 fold expressions work out of the box.

**Comment by hzhangxyz** [comment]
Also happened in cling 0.7
```

****************** CLING ******************
* Type C++ code and press enter to run it *
*             Type .q to exit             *
*******************************************
[cling]$ #include<iostream>
[cling]$ #include<tuple>
[cling]$ auto t = std::make_tuple("hi",3);
[cling]$ auto [a,b] = t;
[cling]$ std::cout << a << "\n";
input_line_7:2:15: error: use of undeclared identifier 'a'
 std::cout << a << "\n";
              ^
```

I start cling by `cling -std=c++17`

**Comment by EdwardWongHauPepeluTivruskiIV** [comment]
Same issue with cling 0.9

**Comment by ferdymercury** [comment]
Current master leads to an assert:

`root.exe: /opt/root_src/interpreter/llvm-project/clang/lib/CodeGen/CodeGenModule.cpp:3350: void clang::CodeGen::CodeGenModule::EmitGlobal(clang::GlobalDecl): Assertion `VD->isFileVarDecl() && "Cannot emit local var decl as global."' failed.`

**Comment by ThePirate42** [comment]
The code above seems to work, but sometimes I get problems when gClingOpts->AllowRedefinition is true and I redefined something that is connected to the tuple (before creating the tuple).
